### PR TITLE
Fix: APP-1636

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Automated fix for proposal dates add

Task: [APP-1636](https://aragonassociation.atlassian.net/browse/APP-1636)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1636]: https://aragonassociation.atlassian.net/browse/APP-1636?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by RakeshUP** [comment]
@sepehr2github, there's something wrong with the minEndDateTimeMills calculation. For the newly created proposal, the end date is off by the gap of 1 hour I set

**Comment by sepehr2github** [comment]
> @sepehr2github, there's something wrong with the minEndDateTimeMills calculation. For the newly created proposal, the end date is off by the gap of 1 hour I set

How do you mean, It works fine for me? Could you explain more??

**Comment by RakeshUP** [comment]
Hey @sepehr2github, providing some screenshots here explaining what I said.

This is my proposal review screen
Screenshot 2023-02-06 at 2 47 47 PM

And this is my proposal details page
Screenshot 2023-02-06 at 2 49 11 PM

I set startTime as 2:47 and clicked create proposal at 2:48. The startTime has changed correctly, but the endTime has shifted by a random number

File: packages/web-app/src/context/createProposal.tsx [hunk 1]
```diff
@@ -276,6 +281,36 @@ const CreateProposalProvider: React.FC<Props> = ({
...
+        if (endDateTime.valueOf() < minEndDateTimeMills) {
+          // Update the endDate + 30 sec for transaction time
+          const legacyStartDate = new Date(
+            `${startDate}T${startTime}:00${getCanonicalUtcOffset(startUtc)}`
+          );
+          const endMills =
+            endDateTime.valueOf() +
+            (startDateTime.valueOf() - legacyStartDate.valueOf()) +
+            // 10 minutes in milliseconds
+            600000;
```
Comment (RakeshUP) [inline packages/web-app/src/context/createProposal.tsx]: This is causing an extra 10 minutes to be added. If I set the proposal duration to be 1 hour, it's getting changed to 70 minutes

**Review by RakeshUP — APPROVED** [review]
Could you solve that extra 10 minutes being added to the end time? Approving this as it's a tiny issue that can be overlooked

File: packages/web-app/src/context/createProposal.tsx [hunk 2]
```diff
@@ -276,6 +282,34 @@ const CreateProposalProvider: React.FC<Props> = ({
...
+        }
+
+        const minEndDateTimeMills =
+          startDateTime.valueOf() +
+          daysToMills(minDays || 0) +
+          hoursToMills(minHours || 0) +
+          minutesToMills(minMinutes || 0);
+
+        if (endDateTime.valueOf() < minEndDateTimeMills) {
+          // Update the endDate + 30 sec for transaction time
```
Comment (WissenIstNacht) [inline packages/web-app/src/context/createProposal.tsx]: I think this comment is no longer valid.

**Review by WissenIstNacht — APPROVED** [review]
Phew that code looks super fragile with the two if statements that both check if `startSwitch === now`. In hindsight you might have been better off rewriting the whole logic in this callback. 

But so far it seems to work fine, so green light from me 🙂 👍🏼

**Pull request merged**